### PR TITLE
Config generator tests for all components

### PR DESCRIPTION
**Pull request by l0kix2**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

First PR for #75

Currently we have test coverage for YTsaurus config generating code only for master and scheduler. Here I added tests for other components, so config generator can be easily refactored if needed.

Changes are only in test code and existing tests behaviour is not changed.

### CODE REVIEW

File: pkg/ytconfig/generator_test.go [hunk 1]
```diff
@@ -14,61 +15,180 @@ import (
...
+		Rack: "fake",
+	}
 )
 
+func TestGetChytInitClusterConfig(t *testing.T) {
+	g := NewGenerator(getYtsaurus(), testClusterDomain)
+	cfg, err := g.GetChytInitClusterConfig()
+	require.NoError(t, err)
+	canonize.Assert(t, cfg)
+}
```
Comment (savnadya) [inline pkg/ytconfig/generator_test.go:75]: Let's add empty lines between functions

File: pkg/ytconfig/generator_test.go [hunk 2]
```diff
@@ -14,61 +15,180 @@ import (
...
+func TestGetDiscoveryConfig(t *testing.T) {
+	g := NewGenerator(withDiscovery(getYtsaurus()), testClusterDomain)
+	cfg, err := g.GetDiscoveryConfig()
+	require.NoError(t, err)
+	canonize.Assert(t, cfg)
+}
+func TestGetExecNodeConfig(t *testing.T) {
+	g := NewGenerator(getYtsaurus(), testClusterDomain)
```
Comment (savnadya) [inline pkg/ytconfig/generator_test.go:95]: Maybe add exec nodes in ytsaurus spec?
Same for data nodes

**Review by savnadya — CHANGES_REQUESTED** [review]

File: pkg/ytconfig/generator_test.go (hunk 2 above)
Comment (l0kix2) [reply pkg/ytconfig/generator_test.go]: As we agreed in chat, I've introduced a new function "ytsaurus spec with every component" and used it in all non-specific tests, so we test generating configs for more realistic spec. 
And for tests that want to test something specific (like master hostnames) we can build spec granularly adding required functions.

**Comment by savnadya** [comment]
Shipped!

**Pull request closed without merging**